src/libopensc/card-muscle.c

```diff
@@ -474,13 +465,8 @@ static int muscle_init(sc_card_t *card)
 
 	card->flags |= SC_CARD_FLAG_RNG;
 	card->caps |= SC_CARD_CAP_RNG;
-
-	/* Card type detection */
-	if (_sc_match_atr(card, muscle_atrs, &card->type) < 0)   {
-		free(priv->fs);
-		free(card->drv_data);
-		SC_FUNC_RETURN(card->ctx, SC_LOG_DEBUG_NORMAL, SC_ERROR_NOT_SUPPORTED);
-	}
+	card->max_recv_size = 255;
+	card->max_send_size = 255;
```

Comment:
please remove this change. Some muscle card indirectly set this data by specifying extended length capabilities

Reply:
See [this comment](https://github.com/OpenSC/OpenSC/pull/1248#discussion_r163543010). I noticed that when card->max_recv_size is setted to 256, PIN structure is not created..